### PR TITLE
FFmpeg conan v2

### PR DESCRIPTION
**Pull request by sorny92**
Specify library name and version:  **ffmpeg/x.x.x**

Updating ffmpeg to conan v2 based on the work of https://github.com/conan-io/conan-center-index/pull/13143.

Also this PR adds support for `5.1`and `5.1.2`as they are the latests versions of the library.
---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/ffmpeg/all/conandata.yml [hunk 1]
```diff
@@ -1,4 +1,10 @@
 sources:
+  "5.1.2":
+    url: "https://ffmpeg.org/releases/ffmpeg-5.1.2.tar.bz2"
+    sha256: "39a0bcc8d98549f16c570624678246a6ac736c066cebdb409f9502e915b22f2b"
+  "5.1":
+    url: "https://ffmpeg.org/releases/ffmpeg-5.1.tar.bz2"
+    sha256: "32b56fb01ce90d452958ae25e91c9564abf49ed5453c127bec23c63e530aa8fa"
```
Comment (MartinDelille) [inline recipes/ffmpeg/all/conandata.yml:7]: I would rather do the conan v2 migration first and then add the version.

File: recipes/ffmpeg/all/conanfile.py [hunk 2]
```diff
@@ -1,14 +1,20 @@
-from conan.tools.files import rename
-from conans import ConanFile, AutoToolsBuildEnvironment, tools
-from conans.errors import ConanInvalidConfiguration
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.apple import is_apple_os, to_apple_arch
+from conan.tools.env import VirtualBuildEnv, VirtualRunEnv
+from conan.tools.build import check_min_cppstd, cross_building
+from conan.tools.files import copy, get, rm, rename, chdir, rmdir, apply_conandata_patches, export_conandata_patches, replace_in_file
+from conan.tools.gnu import Autotools, AutotoolsToolchain, AutotoolsDeps, PkgConfigDeps
```
Comment (MartinDelille) [inline recipes/ffmpeg/all/conanfile.py:7]: You can easily fix linter issue by running it locally (see https://github.com/conan-io/conan-center-index/blob/master/docs/v2_linter.md#running-the-linter-locally)

File: recipes/ffmpeg/all/conanfile.py [hunk 3]
```diff
@@ -531,45 +507,50 @@ def opt_append_disable_if_set(args, what, v):
...
+        tc.generate(env)
+        tc = PkgConfigDeps(self)
+        tc.generate()
+        tc = AutotoolsDeps(self)
+        tc.generate()
+        # inject tools_require env vars in build context
+        ms = VirtualBuildEnv(self)
+        ms.generate(scope="build")
+        #self._autotools.configure(
+        #    args=args, configure_dir=self.source_folder, build=False, host=False, target=False)
```
Comment (MartinDelille) [inline recipes/ffmpeg/all/conanfile.py:553]: You should remove these comments.

File: recipes/ffmpeg/config.yml [hunk 4]
```diff
@@ -1,4 +1,8 @@
 versions:
+  "5.1.2":
+    folder: "all"
+  "5.1":
+    folder: "all"
```
Comment (MartinDelille) [inline recipes/ffmpeg/config.yml:5]: See comment before.

**Review by MartinDelille — CHANGES_REQUESTED** [review]

File: recipes/ffmpeg/all/conanfile.py [hunk 5]
```diff
@@ -179,10 +185,6 @@ class FFMpegConan(ConanFile):
     generators = "pkg_config"
```
Comment (uilianries) [inline recipes/ffmpeg/all/conanfile.py:185]: Is it really required??

File: recipes/ffmpeg/all/conanfile.py [hunk 6]
```diff
@@ -179,10 +185,6 @@ class FFMpegConan(ConanFile):
...
     _autotools = None
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
     @property
     def _is_msvc(self):
```
Comment (uilianries) [inline recipes/ffmpeg/all/conanfile.py:189]: Deprecated, please, use tools.microsoft.is_msvc

File: recipes/ffmpeg/all/conanfile.py [hunk 7]
```diff
@@ -298,7 +312,7 @@ def requirements(self):
             self.requires("vulkan-loader/1.3.221")
 
     def validate(self):
-        if self.options.with_ssl == "securetransport" and not tools.is_apple_os(self.settings.os):
+        if self.options.with_ssl == "securetransport" and not is_apple_os(self):
```
Comment (uilianries) [inline recipes/ffmpeg/all/conanfile.py:315]: validate should use self.info.options

File: recipes/ffmpeg/all/conanfile.py (hunk 5 above)
Comment (sorny92) [reply recipes/ffmpeg/all/conanfile.py:185]: If you remove it you get the next error:
```
ERROR: ffmpeg/5.0@lumicks/cookbook-testing: Error in build() method, line 569
	shutil.copy("x264.pc", "libx264.pc")
	FileNotFoundError: [Errno 2] No such file or directory: 'x264.pc'
```

**Pull request closed without merging**